win32/devdoc/gballoc_ll_jemalloc.md

```diff
@@ -165,4 +167,28 @@ static void jemalloc_print_stats_callback(void* context, const char* text)
 
 **SRS_GBALLOC_LL_JEMALLOC_01_009: [** If `text` is NULL, `jemalloc_print_stats_callback` shall return. **]**
 
-**SRS_GBALLOC_LL_JEMALLOC_01_010: [** Otherwise, `jemalloc_print_stats_callback` shall print (log) `text`, breaking it does in chunks of `LOG_MAX_MESSAGE_LENGTH / 2`. **]**
\ No newline at end of file
+**SRS_GBALLOC_LL_JEMALLOC_01_010: [** Otherwise, `jemalloc_print_stats_callback` shall print (log) `text`, breaking it does in chunks of `LOG_MAX_MESSAGE_LENGTH / 2`. **]**
+
+### gballoc_ll_set_decay
+
+```c
+MOCKABLE_FUNCTION(, int, gballoc_ll_set_decay, int64_t, decay_milliseconds);
+```
+
+`gballoc_ll_set_decay` sets the dirty and muzzy pages decay time to `decay_milliseconds` milliseconds.
```

Comment:
>sets the dirty and muzzy pages decay time to `decay_milliseconds` milliseconds. [](http://example.com/codeflow?start=23&length=79)

One would argue that decay_milliseconds should be 
1) part of "runtime" API (and that happens here)
2) part of "init" API (gballoc_ll_init takes a void* and this value should be behind that void*).

Because what happens with all the allocates between

gballoc_ll_init and
gballoc_ll_set_decay?

If `decay_milliseconds` exists at the time of gballoc_ll_init then it should be used by gballoc_ll_init.

Reply:
The allocations in between would happen as is but once the decay values are set they will just be held for longer in the dirty and muzzy pool rather than being released immediately. So decay values can actually be set before init as well since its an arena level parameter. However, if we plan to pass the decay value to gballoc_hl/ll_init, do you want me to have the same value passed to both dirty and muzzy decay options? We have now separated both the calls. Or should we stick to calling the set_option later?